opensearch-operator/pkg/builders/cluster.go

```diff
@@ -864,6 +865,70 @@ func STSInNodePools(sts appsv1.StatefulSet, nodepools []opsterv1.NodePool) bool
 	return false
 }
 
+func NewSnapshotconfigUpdateJob(
+	instance *opsterv1.OpenSearchCluster,
+	jobName string,
+	namespace string,
+	checksum string,
+	volumes []corev1.Volume,
+	volumeMounts []corev1.VolumeMount,
+) batchv1.Job {
+	httpPort, _ := helpers.VersionCheck(instance)
+	dns := DnsOfService(instance)
+	var snapshotCmd string
+	for _, repository := range instance.Spec.General.Snapshot {
+		var snapshotSettings string
+		for settingsKey, settingsValue := range repository.Settings {
+			snapshotSettings += fmt.Sprintf("\"%s\": \"%s\" , ", settingsKey, settingsValue)
+		}
+		snapshotSettings = strings.TrimRight(snapshotSettings, " ,")
+		snapshotCmd += fmt.Sprintf("curl -s --fail --show-error -k -u \"$(cat /mnt/admin-credentials/username):$(cat /mnt/admin-credentials/password)\" -X PUT https://%s.svc.cluster.local:%v/_snapshot/%s?pretty -H \"Content-Type: application/json\" -d %c{\"type\": \"%s\", \"settings\": {%s}}%c; ", dns, fmt.Sprint(httpPort), repository.Name, '\'', repository.Type, snapshotSettings, '\'')
+	}
+	terminationGracePeriodSeconds := int64(5)
+	backoffLimit := int32(0)
+
+	node := opsterv1.NodePool{
+		Component: "snapshotconfig",
+	}
+	annotations := map[string]string{
+		snapshotconfigChecksumAnnotation: checksum,
+	}
+	image := helpers.ResolveImage(instance, &node)
```

Comment:
Wouldn't it be better to use the inithelper image here instead of a full-blown opensearch?

Reply:
Hey @swoehrl-mw the default image we support `public.ecr.aws/opsterio/busybox:1.27.2-buildx` does not have `curl` installed and this could be the same with user passed init image, hence used the default OpenSearch image.